verilog/formatting/tree_unwrapper.cc

```diff
@@ -3123,6 +3131,36 @@ void TreeUnwrapper::ReshapeTokenPartitions(
       break;
     }
 
+    case NodeEnum::kRegisterVariable:
+    case NodeEnum::kInstantiationType: {
+      size_t partition_size = partition.Children().size();
+      if (partition_size >= 4) {
+        auto subnode = verible::GetSubtreeAsNode(node, tag, 0);
+        if (!subnode) {
+          subnode = verible::GetSubtreeAsNode(node, tag, 1);
+          if (!subnode || subnode->Tag().tag !=
+                              static_cast<int>(NodeEnum::kUnpackedDimensions)) {
+            break;
+          }
+        } else {
+          subnode = verible::GetSubtreeAsNode(*subnode, subnode->Tag().tag, 3);
+          if (!subnode || subnode->Tag().tag !=
+                              static_cast<int>(NodeEnum::kPackedDimensions)) {
+            break;
+          }
+        }
```

Comment:
I think this could be a bit more readable if we extracted either of the dimension nodes first, and only after that checked if `subnode` is `kUnpackedDimensions` or `kPackedDimensions`. That gives one fewer `if` and less nesting.

Also I think the Verible idiom is to do `NodeEnum(subnode->Tag().tag) != NodeEnum::kUnpackedDimensions` instead of the `static_cast`. I know this is not always the case, but personally I find that easier to parse.